ginisdk/src/androidTest/java/net/gini/android/SdkIntegrationTestAccounting.java

```diff
@@ -0,0 +1,361 @@
+package net.gini.android;
+
+import static net.gini.android.helpers.TrustKitHelper.resetTrustKit;
+
+import android.content.Context;
+import android.content.res.AssetManager;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.os.Build;
+import android.support.test.filters.LargeTest;
+import android.support.test.filters.SdkSuppress;
+import android.test.AndroidTestCase;
+import android.util.Log;
+
+import com.android.volley.toolbox.NoCache;
+
+import net.gini.android.DocumentTaskManager.DocumentUploadBuilder;
+import net.gini.android.authorization.EncryptedCredentialsStore;
+import net.gini.android.authorization.UserCredentials;
+import net.gini.android.helpers.TestUtils;
+import net.gini.android.models.Document;
+import net.gini.android.models.SpecificExtraction;
+
+import org.json.JSONException;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import bolts.Continuation;
+import bolts.Task;
+
+@LargeTest
+public class SdkIntegrationTestAccounting extends AndroidTestCase {
+
+    private Gini gini;
+    private String clientId;
+    private String clientSecret;
+    private String apiUriAccounting;
+    private String userCenterUri;
+
+    @Override
+    protected void setUp() throws Exception {
+        final AssetManager assetManager = getContext().getResources().getAssets();
+        final InputStream testPropertiesInput = assetManager.open("test.properties");
+        assertNotNull("test.properties not found", testPropertiesInput);
+        final Properties testProperties = new Properties();
+        testProperties.load(testPropertiesInput);
+        clientId = getProperty(testProperties, "testClientId");
+        clientSecret = getProperty(testProperties, "testClientSecret");
+        apiUriAccounting = getProperty(testProperties, "testApiUriAccounting");
+        userCenterUri = getProperty(testProperties, "testUserCenterUri");
+
+        Log.d("TEST", "testClientId " + clientId);
+        Log.d("TEST", "testClientSecret " + clientSecret);
+        Log.d("TEST", "testApiUriAccounting " + apiUriAccounting);
+        Log.d("TEST", "testUserCenterUri " + userCenterUri);
+
+        resetTrustKit();
+
+        gini = new SdkBuilder(getContext(), clientId, clientSecret, "example.com").
+                setApiBaseUrl(apiUriAccounting).
+                setGiniApiType(GiniApiType.ACCOUNTING).
+                setUserCenterApiBaseUrl(userCenterUri).
+                setConnectionTimeoutInMs(60000).
+                build();
+    }
+
+    private static String getProperty(Properties properties, String propertyName) {
+        Object value = properties.get(propertyName);
+        assertNotNull(propertyName + " not set!", value);
```

Comment:
Same here. This should fail in the test itself.

Reply:
`setUp()` is called before each test. Practically it is part of every test. It's just how the code is organized in order to avoid duplication.

By having the asserts here the tests fail early and one knows exactly what's wrong.